### ISSUE TITLE
Accounts.config: `sendVerificationEmail` doesn't work, if user is created on the server.

### ISSUE THREAD

**Issue by rmuratov** [meteor/meteor#8694]
If I forbid accounts creation from the client, I have to create users through methods. In this case, I expect verification email will be anyway sent to the new user, but it doesn't work.

Linux Mint 18, Meteor 1.4.4.2

```shell
$ meteor create test
$ cd test
$ meteor add accounts-base accounts-password
$ meteor npm install --save bcrypt
$ meteor
```

Then for the sake of the test, I create file `both/both.js`:

```js
Accounts.config({
  sendVerificationEmail: true
});
```

Then I go to browser console and run:

```js
Accounts.createUser({ email: 'mail@mail.com', password: 'pass'});
```

Since I didn't specified `MAIL_URL`, on the server log I see attempt to send email:

```
====== BEGIN MAIL #0 ======
...
====== END MAIL #0 ======
```

Then I run the same code through `meteor shell` (or in my actual method) and I don't see any attempts to send a verification email.

I think at least this behavior should be mentioned in the [related docs part](https://docs.meteor.com/api/accounts-multi.html#AccountsCommon-config).

**Comment by hwillson** [comment]
Hi @rmuratov - this should be working properly. I created and tested a repro following your steps (https://github.com/hwillson/meteor-issue-8694), and everything is working as it should. Are you sure you aren't expecting the console email dump to happen a second time, for the same email address? The verification email will only be sent once if the user has a valid email verification token associated with their record (which happens after the first `Accounts.createUser` call). Try a new email address like `mail1@mail.com`. I'll close this for now as it appears to be working properly, but please post back if you're still encountering this issue (ideally with a runnable reproduction). Thanks!

**Comment by macrozone** [comment]
i can confim that this does not work.

Users created on the server will not receive a verification email.

tested on meteor 1.8

**Comment by klaussner** [comment]
I was able to reproduce this with the repository linked above (https://github.com/hwillson/meteor-issue-8694):

1. Clone the repo.
2. Update to the latest Meteor version.
3. Run the app.
4. Open a Meteor shell and run:
```
Accounts.createUser({ email: "from-server@example.com", password: "pw" });
```
5. Open the app in a browser and run:
```
Accounts.createUser({ email: "from-client@example.com", password: "pw" });
```

Only the second call (from the client) will log a `====== BEGIN MAIL #0 ====== ...` message on the server.

The problem seems to be that calling `Accounts.createUser` on the server executes [this `createUser` function](https://github.com/meteor/meteor/blob/af26e8b052a5135033e561cf4e4347eee585ab3b/packages/accounts-password/password_server.js#L1050), whereas calling it on the client executes [this Meteor method](https://github.com/meteor/meteor/blob/af26e8b052a5135033e561cf4e4347eee585ab3b/packages/accounts-password/password_server.js#L1093), which calls the `createUser` function *and* `Accounts.sendVerificationEmail`.

**Comment by fidelsam1992** [comment]
Any updates on this issue ? I am facing the same

**Comment by cecilcosta** [comment]
A'm facing the same issue as well.

**Comment by sebakerckhof** [comment]
I think this behavior makes sense and is in line with the docs: https://docs.meteor.com/api/passwords.html#Accounts-createUser
The docs could be made more clear though.
Here's the reasoning:
- If you create an account on the server, the input doesn't necessarily come from the server (could be by some admin action / import / ...) and in those cases it doesn't make sense to send the verification mail.
- As the docs mention, on the server you don't need to specify a password. In that case you will want to send an enrollment email ( Accounts.sendEnrollmentEmail ). In this case you will also not want to have both the enrollment email and the verification email.

You can always call `Accounts.sendVerificationEmail(userId, options.email);` yourself on the server.

But this is indeed one of the cases where meteor's Isomorphism breaks, which is confusing to new users and maybe we can find a way to improve this. However, many apps may rely on the current behavior and this may be a change which is not immediately visible.

So for now I would just make this more clear in the docs.

**Comment by cstrat** [comment]
I am running into this issue now, not sure why it hasn't come up before.

I can't even use `Accounts.sendVerificationEmail()` in the `Accounts.onCreateUser()` hook because the user doesn't exist yet. I saw a workaround in the forums by using a timeout... seems silly?

**Comment by filipenevola** [comment]
Hi, I believe we should update in the docs these quirks and possible solutions.

**Comment by delki8** [comment]
It sounds like an easy problem to solve but the design decisions may be somewhat controversial. I thought about some and would like to hear what y'all think:

1. add a new boolean arg to [`createUser()`](https://github.com/meteor/meteor/blob/devel/packages/accounts-password/password_server.js#L1091) and move the [send email verification](https://github.com/meteor/meteor/blob/devel/packages/accounts-password/password_server.js#L1160) from the method into the `createUser()` function. This could also be done using the `options` object that `createUser()` already uses. I couldn't think of a good name to an arg like this; `enforcesEmailVerification` was the best I could find.
2. write a new function `createUserEnforcingEmailVerification()` or something like it. This function would work as a replacement for the current `createUser()`. It would be called by the [current method](https://github.com/meteor/meteor/blob/devel/packages/accounts-password/password_server.js#L1150) and could be called by anyone wishing to send the verification email even if the call doesn't come from the client.
3. add a new option to the Accounts package similar to `sendVerificationEmail`, called `enforceEmailVerification` or something. This would make `createUser()` always send an email verification no matter from where it was being called, client or server.
4. other?
5. none?

I think none of them would raise backward compatibility issues. Number 2 is my personal favorite. I know they all have drawbacks but I'm not particularly inspired to walk through them right now, so let's talk.

**Comment by filipenevola** [comment]
It will be released with Meteor 1.12
